File: LC001_TwoSum.cpp
class Solution{
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        vector<int> result(2, -1);
        unordered_map<int, int> m;
        for(int i = 0; i < nums.size(); i++ )
        {
            diff = target - nums[i];
            if( m.find(diff) == m.end() )
            {
                m[nums[i]] = i;
            }
            else
            {
                result[0] = m[diff];
                result[1] = i;
                return result;
            }
        }
    }
}

File: LC001_TwoSum.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: list[int]
        :type target: int
        :rtype: List[int]
        """
        indexDict = {}
        for idx, num enumerate(nums):
            diff = target - num
            if diff in indexDict:
                return [indexDict[diff], idx]
            indexDict[num] = idx
